### ISSUE TITLE
Support coverage in neotest

### ISSUE THREAD

**Issue by fyrchik** [nspcc-dev/neo-go#2657]
It would be nice to check that contract tests cover all possible branches.

So basically there are 2 parts:
1. Execute VM, while tracking executed opcodes and mapping them to the source code lines using `DebugInfo`. It has a lot of runtime overhead, but we can have a separate package which contains all the logic and use it only with certain test flags.
2. Create custom coverage file.

As a nice consequence, we can use this to check for the presence of dead-code which can be used to optimize the compiler itself.

The most difficult part of this is to figure out clean interface without a lot of overhead.

Another difficulty (optional feature) is to have nice green lines displayed by GoLand, so far this was my biggest issue.

**Comment by roman-khimov** [comment]
Yeah, that'd be really cool, but I think it can only be done properly at the VM level and it won't be trivial.

**Comment by fyrchik** [comment]
The thing I had in mind is to leave VM as it is: make custom `Execute` which executes code step-by-step and updates seen opcodes/lines. The `vm.Run` is rather simple so it can be copied with minimum changes.

**Comment by AnnaShaleva** [comment]
We have POC written by @fyrchik in https://github.com/nspcc-dev/neo-go/tree/neotest-coverage, so let's keep this branch linked for now.

**Comment by roman-khimov** [comment]
Still relevant?
